Clock.py

```diff
@@ -30,61 +30,134 @@
+
+#
+# Convert seconds since midnight to the corresponding clock minute
+#
+def clock_minutes(t):
+    return int(int(t % 3600) / 60)
+
+#
+# Take an starting point and round up to the next interval-multiple:
+#
+def round_up(start, interval):
+    return start - (start % interval) + interval # Round up to next interval (sec)
+
+#
+# Generate a proper time announcement:
+#
+#   "midnight"
+#   "Nine o"clock"
+#   "A quarter past nine"
+#   "9:20"
+#   "Half past nine"
+#   "A quarter to ten"
+#   "noon"
+#
+def announcement(hours, minutes):
+    msg = "The time is "
+    if minutes == 0:
+        if hours == 0:
+            msg += "midnight"
+        elif hours == 12:
+            msg += "noon" + 12 * "L "
+        else:
+            msg += number_to_text(truncate_hours(hours)) + " o'clock     " + truncate_hours(hours) * "L "
+    elif minutes == 15:
+        msg += "a quarter past " + number_to_text(truncate_hours(hours))
+    elif minutes == 30:
+        msg += "half past " + number_to_text(truncate_hours(hours))
+    elif minutes == 45:
+        msg += "a quarter to " + number_to_text(truncate_hours(hours + 1))
+    else:
+        # Last resort: just generate the time in hh:mm format
+        msg += "{hour}:{minute:02d}".format(hour=hours, minute=minutes)
+    return msg
+
+#
+# Send a message as morse on the sounder:
+#
 def announce(s, kob, sender):
     for c in s:
         code = sender.encode(c)
         kob.sounder(code)
 
 try:
+    import argparse
     import sys
     import time
     from pykob import config, kob, morse, log
 
-    log.log('Starting Clock')
-
-    PORT = config.Port # serial port for KOB interface
-    WPM = config.Speed  # code speed (words per minute)
-    SOUND = config.Sound # whether to enable computer sound for sounder
-    MSGS = [
-        ( 900, 'The time is nine oclock     L  L  L  L  L  L  L  L  L'),
-        (1000, 'The time is ten oclock     L  L  L  L  L  L  L  L  L  L'),
-        (1100, 'The time is eleven oclock     L  L  L  L  L  L  L  L  L  L  L'),
-        (1200, 'The time is twelve oclock     L  L  L  L  L  L  L  L  L  L  L  L'),
-        (1300, 'The time is one oclock     L'),
-        (1400, 'The time is two oclock     L  L'),
-        (1500, 'The time is three oclock     L  L  L'),
-        (1600, 'The time is four oclock     L  L  L  L'),
-        (1700, 'The time is five oclock     L  L  L  L  L'),
-        (1800, 'The time is six oclock     L  L  L  L  L  L'),
-        (1900, 'The time is seven oclock     L  L  L  L  L  L  L'),
-        (2000, 'The time is eight oclock     L  L  L  L  L  L  L  L'),
-        (2100, 'The time is nine oclock     L  L  L  L  L  L  L  L  L'),
-        (2200, 'The time is ten oclock     L  L  L  L  L  L  L  L  L  L')]
-
-    myKOB = kob.KOB(port=PORT, audio=SOUND)
-    mySender = morse.Sender(WPM)
+    log.log("Starting Clock")
+
+    clock_parser = argparse.ArgumentParser(parents=[config.PortOverride, config.SoundtOverride, config.WPMOverride])
```

Comment:
So, does every program that wants to use arguments other than the 'standard' arguments have to do this? 
I guess it's not too bad if we document it well.

Reply:
I think of it as "Here's how you specify the standard, pre-configurable parameters that you want to be able to override".  You can omit overrides for things that don't apply to avoid getting into a situation where you end up accepting overrides for configuration parameters that don't even apply to the particular program?